polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -5155,8 +5161,13 @@ void MLIRASTConsumer::run() {
     if (done.count(name))
       continue;
 
-    LLVM_DEBUG(llvm::dbgs() << "\n-- FUNCTION BEING EMITTED: "
-                            << FD->getNameAsString() << " --\n\n";);
+    LLVM_DEBUG({
+      bool isDeviceFuncOrKernel = FD->hasAttr<clang::SYCLDeviceAttr>() ||
+                                  FD->hasAttr<clang::SYCLKernelAttr>();
+      llvm::dbgs() << (isDeviceFuncOrKernel ? "\n -- DEVICE " : "\n -- HOST ");
+      llvm::dbgs() << "FUNCTION BEING EMITTED: " << FD->getNameAsString()
+                   << " --\n\n";
+    });
```

Comment:
This will be super handy for debugging :+1:, but I'd rather move this call after the call to `GetOrCreateMLIRFunction` and use the `Context` member of the `FunctionToDeclare` struct. That is what really points out where the function is going to be emitted.

Reply:
This draft PR has all changes from https://github.com/intel/llvm/pull/6887 (by Ettore), I believe he will put this change in https://github.com/intel/llvm/pull/6916 😃